app/Services/Services/Sharing/ServiceOptionExporterService.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Pterodactyl - Panel
+ * Copyright (c) 2015 - 2017 Dane Everitt <dane@daneeveritt.com>.
+ *
+ * This software is licensed under the terms of the MIT license.
+ * https://opensource.org/licenses/MIT
+ */
+
+namespace Pterodactyl\Services\Services\Sharing;
+
+use Carbon\Carbon;
+use Pterodactyl\Contracts\Repository\ServiceOptionRepositoryInterface;
+
+class ServiceOptionExporterService
+{
+    /**
+     * @var \Carbon\Carbon
+     */
+    protected $carbon;
+
+    /**
+     * @var \Pterodactyl\Contracts\Repository\ServiceOptionRepositoryInterface
+     */
+    protected $repository;
+
+    /**
+     * XMLExporterService constructor.
+     *
+     * @param \Carbon\Carbon                                                     $carbon
+     * @param \Pterodactyl\Contracts\Repository\ServiceOptionRepositoryInterface $repository
+     */
+    public function __construct(
+        Carbon $carbon,
```

Comment:
I disagree with injecting non-services or non-repositories. Because Laravel hard codes Carbon, you might as well too. The extra effort to inject it won't matter because it's not exactly replaceable in any sense either way.

Reply:
I agree with this as well, mostly in regards to Carbon since it has easy testing abilities. I think this might be something to address in a future PR though, as the currently design does DI for most everything.

Reply:
@schrej I believe he was referring to the effort involved human-wise.

Reply:
Yes. There's no benefits to injecting it via the constructor when there's already the Laravel Carbon Facade (and no Carbon service or repository).